Tools/Scripts/libraries/webkitcorepy/webkitcorepy/file_lock.py

```diff
@@ -22,60 +22,86 @@
 # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 import os
+import re
 import sys
 import time
 
+from webkitcorepy import string_utils
+
 if sys.platform.startswith('win'):
     import msvcrt
-else:
-    import fcntl
 
 
 class FileLock(object):
+    INTEGER_RE = re.compile(r'^\d+$')
+    USE_WINDOWS = sys.platform.startswith('win')
+    USE_EXLOCK = not USE_WINDOWS and getattr(os, 'O_EXLOCK', False)
+
+    @classmethod
+    def is_process_running(cls, pid):
+        try:
+            os.kill(pid, 0)
+            return True
+        except OSError:
+            return False
 
     def __init__(self, path, timeout=20):
         self.path = path
         self.timeout = timeout
         self._descriptor = None
+        self._depth = 0
 
     @property
     def acquired(self):
         return bool(self._descriptor)
 
     def acquire(self):
         if self._descriptor:
+            self._depth += 1
```

Comment:
Not super happy about making this a recursive lock, as that's generally an anti-pattern. Is this strongly needed?